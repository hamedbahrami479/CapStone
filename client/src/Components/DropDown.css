.section {

    justify-content: center;
    display: block;
    font-size: large;

}