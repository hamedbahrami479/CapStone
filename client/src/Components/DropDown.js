import React, { useState } from 'react';

function DropDown() {
  const [product, setProduct] = useState();
  return (
    <div>
      <select
        className="custom-select"
        id="SelectID"
        value={product}
        onChange={e => setProduct(e.target.value)}
      >
        <option>Select an Option</option>
        <option>Plastic Transparent</option>
        <option>Plastic Black</option>
        <option>Aluminum</option>
        <option>Aluminum Contaminated</option>
        <option>Paper Contaminated</option>
        <option>Paper Coated</option>
        <option>Paper Clean</option>
      </select>
    </div>
  );
}

export default DropDown;
