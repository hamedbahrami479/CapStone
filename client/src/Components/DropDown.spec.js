import React from 'react';
import { render, screen } from '@testing-library/react';
import DropDown from './DropDown'

describe("Test for DropDown Component", () => {
    it("snapshot DropDown component", () => {
        const {asFragment} = render(<DropDown/>);
        expect(asFragment(DropDown)).toMatchSnapshot();
});

it("should have the text as defined in the DropDown Component", () => {
    const {SelectorMatcherOptions} = render(<DropDown/>)
    expect(screen.getByText(/Plastic Transparent/i)).toHaveTextContent('Plastic Transparent');
    })
});