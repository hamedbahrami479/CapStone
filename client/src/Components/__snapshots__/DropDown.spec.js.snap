// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for DropDown Component snapshot DropDown component 1`] = `
<DocumentFragment>
  <div>
    <select
      class="custom-select"
      id="SelectID"
    >
      <option>
        Select an Option
      </option>
      <option>
        Plastic Transparent
      </option>
      <option>
        Plastic Black
      </option>
      <option>
        Aluminum
      </option>
      <option>
        Aluminum Contaminated
      </option>
      <option>
        Paper Contaminated
      </option>
      <option>
        Paper Coated
      </option>
      <option>
        Paper Clean
      </option>
    </select>
  </div>
</DocumentFragment>
`;
